yt_dlp/extractor/educast.py

```diff
@@ -0,0 +1,455 @@
+from .common import InfoExtractor
+from ..networking import HEADRequest
+from ..utils import (
+    float_or_none,
+    int_or_none,
+    mimetype2ext,
+    str_or_none,
+    traverse_obj,
+    unified_timestamp,
+)
+
+
+class EducastIE(InfoExtractor):
+    _VALID_URL = r'https?://(www)?educast\.fccn\.pt/vod/clips/(?P<id>[a-zA-Z0-9]+)'
+    _API_BASE = 'https://educast.fccn.pt'
```

Comment:
add a baseclass, move `_API_BASE` and `_paginate_and_collect` into it, and have the IE classes inherit from it